### ISSUE TITLE
option -mn turns '&' into ';'

### ISSUE THREAD

**Issue by step-** [mvdan/sh#713]
```sh
# shfmt --version
v3.3.0 # linux 64
```
Reproduce error:
```sh
# cat | shfmt -mn
(
:
) &
# end
^D
```
output:
```sh
(:) \
;
```

All is well when option `-mn` isn't used
```sh
# cat | shfmt
(
:
) &
# end
^D
```
output:
```sh
(
        :
) &
# end
```

Sometimes - but not all the times, depending on code complexity - I can get `-mn` to behave correctly if the line before `) &` is continued with a backslash.

Thank you

**Comment by mvdan** [comment]
Dup of #710 perhaps?

**Comment by step-** [comment]
> Dup of #710 perhaps?

Hmm, they look different.  #710 is about `& }` while #713 is about `) &`.  They could be related, though.  Also, I can't confirm that adding a backslash is a fix-all work-around.  It depends: in my tests backslash sometimes helped but other times it didn't.

**Comment by mvdan** [comment]
I can confirm the issue reproduces with v3.3.0, but not on master after the fix for #710, so it does look like the issue is a duplicate in the sense that it got fixed as part of #710 :)

There have been a few bugfixes in master since that release, so it's likely I'll do v3.3.1 in a week or so. Until then, you can consider running master, which should be pretty easy if you have Go 1.16.x installed: `go install mvdan.cc/sh/v3@master`.

**Comment by step-** [comment]
Thank you!
